### PR TITLE
Add If-Match and If-Not-Match headers when putting files

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@PVince81 ready for review - THX

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
+  only:
+  - master
+
+env:
+  - OC_HASH: bb8b8878b40edaeeba831663275d9815272cdea2
+
 before_install:
-  - docker pull owncloud:$OC_VERSION
-  - DOCKER_ID=$(docker run -d -p 8888:80 owncloud:$OC_VERSION)
+  - wget https://github.com/owncloud/core/archive/$OC_HASH.zip
```
Comment (PVince81) [inline .travis.yml]: will we use dailies in the future ?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM owncloud/base:bionic
+
+LABEL maintainer="ownCloud DevOps <devops@owncloud.com>" \
+  org.label-schema.name="ownCloud Server" \
+  org.label-schema.vendor="ownCloud GmbH" \
+  org.label-schema.schema-version="1.0"
+
+ADD owncloud-*.tar.bz2 /var/www/
+
+RUN find /var/www/owncloud \( \! -user www-data -o \! -group root \) -print0 | xargs -r -0 chown www-data:root && \
```
Comment (PVince81) [inline Dockerfile]: don't we have an official docker that already does this ?

File: .travis.yml (hunk 1 above)
Comment (DeepDiver1975) [reply .travis.yml]: yes

File: tests/appsTest.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 describe('Main: Currently testing apps management,', function () {
 // CURRENT TIME
-  var timeRightNow = new Date().getTime()
+  var timeRightNow = Math.random().toString(36).substr(2, 9)
```
Comment (PVince81) [inline tests/appsTest.js]: better use hardcoded values for deterministic results

File: Dockerfile (hunk 2 above)
Comment (DeepDiver1975) [reply Dockerfile]: Not needed - will drop this file ...

File: tests/appsTest.js (hunk 3 above)
Comment (DeepDiver1975) [reply tests/appsTest.js]: The idea is to add a suffix on each test run so that files and folders don't collide when rerunning the test suite.

**Comment by DeepDiver1975** [comment]
> raise ticket for using official docker ?

I'm using an official docker image - the Dockerfile is a left-over ....

File: tests/appsTest.js (hunk 3 above)
Comment (PVince81) [reply tests/appsTest.js]: ah ok

**Pull request merged**